### DISCUSSION TITLE
Run Viewer without BrowserAuthorizationclient

### DISCUSSION THREAD

**Discussion by hussainshahzad5471** [iTwin/community#45] [Q&A] [answered]
The Viewer requires BrowserAuthorizationClient, I want to be able to run the viewer using just a bearer token using an api only and without interaction with the browser.

      <Viewer
        authClient={authClient}

**Comment by aruniverse** [comment]
Hi, please take a look at this [example](https://github.com/calebmshafer/itwin-viewer-proxy)

**Comment by hussainshahzad5471** [comment]
[itwinv4.log](https://github.com/user-attachments/files/15751666/itwinv4.log)
[package.json](https://github.com/user-attachments/files/15751668/package.json)
Hi Arun, 
I tried the example, first it asks me to move to Version 4 and when i change the package.json to use the 4.x components then i get a ton of errors in the console log, also the application fails to compile because of:
npm ERR! path C:\Users\hussa\source\repos\AWS\iTwinApp\itwin-viewer-proxy-main\react-viewer\@itwin\itwinui-illustrations-react\Svg404/package.json
Failed to compile.

./node_modules/@itwin/core-react/node_modules/@itwin/itwinui-react/esm/core/NonIdealState/ErrorPage.js
Module not found: Can't resolve '@itwin/itwinui-illustrations-react/Svg401' in 'C:\Users\hussa\source\repos\AWS\iTwinApp\itwin-viewer-proxy-main\react-viewer\node_modules\@itwin\core-react\node_modules\@itwin\itwinui-react\esm\core\NonIdealState'


I have attached the logs and package.json for your reference.

**Reply by aruniverse** [reply]
Did you add a dependency on `@itwin/itwinui-illustrations-react` ?
fyi @GerardasB

**Reply by hussainshahzad5471** [reply]
yes, I ran 
npm install @itwin/itwinui-illustrations-react --force
then when i try to build or start i get the below in the terminal:

npm run build
> viewer@0.3.0 build
> react-scripts --openssl-legacy-provider build
Creating an optimized production build...
Failed to compile.
.\node_modules\@itwin\core-react\node_modules\@itwin\itwinui-react\esm\core\NonIdealState\ErrorPage.js
Cannot find module: '@itwin/itwinui-illustrations-react/Svg401'. Make sure this package is installed.
You can install this package by running: npm install @itwin/itwinui-illustrations-react/Svg401.

**Reply by aruniverse** [reply]
Youre going to need to provide more details so we can be more helpful.
What version of itwin/core-react are you using?

**Reply by hussainshahzad5471** [reply]
Hi @aruniverse , 
I just ran a npm audit fix --force, and the viewer is working now, let me check further and provide more details.

**Reply by GerardasB** [reply]
illustrations package should have been installed together with `@itwin/appui-react` https://github.com/iTwin/appui/blob/265f8ab97acbaf59ad0d63bbf1385325459c7ddc/ui/appui-react/package.json#L120

**Comment by hussainshahzad5471** [comment]
Hi @aruniverse, @GerardasB,
when i run the sample you have given earlier in the discussion [sample](https://github.com/calebmshafer/itwin-viewer-proxy)
that is working fine. but when i integrate that code to my application, i get the error :
"scripts/parse-imdl-worker.js:1  Uncaught SyntaxError: Unexpected token '<'" , i dont know where to look for this .js file.
"@itwin/core-react" version is  "^4.13.4", 
i have attached the working json and the not working json for your reference, please let me know if you need anything else.
[package_not_working.json](https://github.com/user-attachments/files/15783867/package_not_working.json)
[package_json_working.json](https://github.com/user-attachments/files/15783869/package_json_working.json)

**Reply by hussainshahzad5471** [reply]
Hi @aruniverse ,
any suggestion on this issue ?

**Reply by aruniverse** [reply]
Do you have a Github repo where you can reproduce this issue?

**Comment by contactshahzadh1** [comment]
Hi @aruniverse,
I was able to run this program somewhere in June last year, is this code still supported?

Now i am getting this error on the landing page :
<img width="946" alt="image" src="https://github.com/user-attachments/assets/bfa42562-5efb-448b-bbab-8f379268d9b5" />

It still uses the old dependencies:
 "@bentley/icons-generic": "^1.0.13",
    "@itwin/core-telemetry": "^3.0.0",
    "@itwin/appui-abstract": "^3.0.0",
    "@itwin/appui-layout-react": "^3.0.0",
    "@itwin/appui-react": "^3.0.0",
    "@itwin/browser-authorization": "^0.5.1",
    "@itwin/components-react": "^3.0.0",
    "@itwin/core-bentley": "^3.0.0",
    "@itwin/core-common": "^3.0.0",
    "@itwin/core-frontend": "^3.0.0",
    "@itwin/core-geometry": "^3.0.0",
    "@itwin/core-i18n": "^3.0.0",
    "@itwin/core-markup": "^3.0.0",
    "@itwin/core-orbitgt": "^3.0.0",
    "@itwin/core-quantity": "^3.0.0",
    "@itwin/core-react": "^3.0.0",
    "@itwin/imodel-components-react": "^3.0.0",
    "@itwin/imodels-access-frontend": "^1.0.1",
    "@itwin/imodels-client-management": "^1.0.1",
    "@itwin/itwinui-react": "^1.16.2",
    "@itwin/presentation-common": "^3.0.0",
    "@itwin/presentation-frontend": "^3.0.0",
    "@itwin/reality-data-client": "^0.7.0",
    "@itwin/web-viewer-react": "^2.0.1",
    "@itwin/webgl-compatibility": "^3.0.0",
    "history": "^4.10.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-redux": "^7.2.0"

**Reply by aruniverse** [reply]
Hi,
As per our [support policy](https://www.itwinjs.org/learning/api-support-policies/#version-support-status), we no longer support iTwin.js 3.x. As such we have shut down the 3.x visualization backend. 
However, the example code once updated to the latest versions of the iTwin.js packages (4.x) it should continue to work again. You can use this [migration guide](https://github.com/iTwin/viewer/blob/master/releases/CHANGELOG-4.0.md) for reference as you update.

**Comment by contactshahzadh1** [comment]
Hi @aruniverse,
I tried like you said, the upgrade was successful, but then i am getting error with this api https://api.bentley.com/imodels/{imodelid}
{
    "error": {
        "code": "iModelNotFound",
        "message": "Requested iModel is not available."
    }
}
Turns out that when i use the token from the 
import { BrowserAuthorizationClient } from "@itwin/browser-authorization"; then i am getting the imodel however the token generated from 
import { ServiceAuthorizationClient } from "@itwin/service-authorization"; gives the above error.
![image](https://github.com/user-attachments/assets/c0deb00a-3ec4-491d-a140-895474de49ac)

![image](https://github.com/user-attachments/assets/299b761f-cb7e-4448-899d-d9b400bd9cff)

**Reply by aruniverse** [reply, marked as answer]
Tokens generated by "@itwin/service-authorization" are tied specifically to the service user account, which look like `<CLIENT_ID>@apps.imsoidc.bentley.com`. Please make sure that service account has been added to the iTwin of the iModel you are trying to access.

**Reply by contactshahzadh1** [reply]
Thanks, this works!

**Comment by contactshahzadh1** [comment]
@aruniverse, @GerardasB,
Could you advise on the above issue as well.
